### DISCUSSION TITLE
How to cope with implicit dependencies?

### DISCUSSION THREAD

**Discussion by sschneider-ihre-pvs** [pkgxdev/pkgx#440] [Help]
Lets take Lunarvim as an example. Lunarvim installscript will look for Neovim like `command -v nvim` if nvim is installed with tea it will not find it since it is not in the PATH. also, if you tick the optional installs it would need nodejs and npm, and python and rust along the way.

How do I make this work with tea?

**Comment by jhheider** [comment]
With an install script, you could do one of the following:
```sh
tea +neovim.io +nodejs.org +npmjs.com +python.org +rust-lang.org _install_script_name_
```
But, more likely what you want is:
```sh
for x in nvim node npm python rust; do
  ln -s $(which tea) /usr/local/bin/$x;
done
_install_script_name_
```

**Reply by sschneider-ihre-pvs** [reply]
That seems really cumbersome -.- and also you have to iteratively figure out the dependencies manually

**Reply by sschneider-ihre-pvs** [reply]
> With an install script, you could do one of the following:
> 
> ```shell
> tea +neovim.io +nodejs.org +npmjs.com +python.org +rust-lang.org _install_script_name_
> ```
> 
> But, more likely what you want is:
> 
> ```shell
> for x in nvim node npm python rust; do
>   ln -s $(which tea) /usr/local/bin/$x;
> done
> _install_script_name_
> ```

Would it make sense to give tea a command for linking/shimming that itself? like something `tea +nodejs.org --link` or so?

**Reply by sschneider-ihre-pvs** [reply]
> With an install script, you could do one of the following:
> 
> ```shell
> tea +neovim.io +nodejs.org +npmjs.com +python.org +rust-lang.org _install_script_name_
> ```
> 
> But, more likely what you want is:
> 
> ```shell
> for x in nvim node npm python rust; do
>   ln -s $(which tea) /usr/local/bin/$x;
> done
> _install_script_name_
> ```

This doesn't seem to work good with parameters, if for example `npm --yes` or `nvim -u` is called

**Reply by mxcl** [reply]
Yes it makes sense and we have plans to add functionality to make this less cumbersome. Being cumbersome is not our thing.

We’ve only been out 4 months, but we’re trying to get to where you want as soon as possible!

> This doesn't seem to work good with parameters, if for example npm --yes or nvim -u is called

We use symlinking for global tools internally at tea for all sorts of things, so it certainly works-for-us. If you can't get it working then please open a ticket and we'll help out or fix things.

**Reply by sschneider-ihre-pvs** [reply]
scoop on windows does the symlink thing also I think, having a current folder symlink to the current used version and having the executable pointing to that folder so there is no need to update that one every time :D

**Comment by mxcl** [comment]
What's the context here?

If lunar vim is packaged with tea then all the dependencies are added to the `env` automatically so lunar vim will find them.

Are you trying to use lunar vim without packaging it for tea as a user or as someone trying to make an install script for lunar vim use tea if it is available?

**Reply by sschneider-ihre-pvs** [reply]
the former, I guess there is no package for that right now. I was more pointing at the problem with having extendable software with unknown dependencies due to plugins that require a general availability of tools and how to generally cope with those situations.
There is also as it seems a context problem with tools like npm where the node_modules in the current folder is not recognized if used with tea for example.

**Reply by sschneider-ihre-pvs** [reply]
I really like the magic of tea, I only have difficulties with managing dependencies of stuff not packaged with tea that also should use stuff that tea has to offer.
